### PR TITLE
[deckhouse-controller] Don't run validation for disabled modules

### PR DESCRIPTION
**Pull request by libmonsoon-dev**
## Description
Validation configs of disabled modules is disabled

## Why do we need it, and what problem does it solve?
https://github.com/deckhouse/deckhouse/issues/7151

## Checklist
- [x] The code is covered by unit tests.
- [x] e2e tests passed.
- [ ] Documentation updated according to the changes.
- [x] Changes were tested in the Kubernetes cluster manually.

## Changelog entries

```changes
section: deckhouse-controller
type: fix
impact_level: default
```

### CODE REVIEW

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @libmonsoon-dev. I've started the workflow for label 'e2e/run/static'...

**Review by nabokihms — APPROVED** [review]

**Comment by miklezzzz** [comment]
it seems it somehow affect module configs' statuses, give me some time to elaborate on it.

**Comment by miklezzzz** [comment]
@libmonsoon-dev could you please also update this part of configs' statuses https://github.com/deckhouse/deckhouse/blob/2862fc05e43d06248f039ef3d81418d0a78eb1de/go_lib/deckhouse-config/status.go#L147-L161 ?
with your changes there is no sense in checking if `IsModuleEnabled` anymore

**Comment by libmonsoon-dev** [comment]
> @libmonsoon-dev could you please also update this part of configs' statuses
> 
> https://github.com/deckhouse/deckhouse/blob/2862fc05e43d06248f039ef3d81418d0a78eb1de/go_lib/deckhouse-config/status.go#L147-L161
> 
> ?
> with your changes there is no sense in checking if `IsModuleEnabled` anymore

Done

**Review by yalosev — APPROVED** [review]

File: go_lib/deckhouse-config/status.go [hunk 1]
```diff
@@ -147,12 +147,7 @@ func (s *StatusReporter) ForConfig(cfg *v1alpha1.ModuleConfig) ModuleConfigStatu
 	if chain.IsKnownVersion(cfg.Spec.Version) && hasVersionedSettings(cfg) {
 		res := Service().ConfigValidator().Validate(cfg)
 		if res.HasError() {
-			var prefix = "Ignored"
-			if s.moduleManager.IsModuleEnabled(cfg.GetName()) {
-				prefix = "Error"
-			}
-
-			invalidMsg := fmt.Sprintf("%s: %s", prefix, res.Error)
+			invalidMsg := fmt.Sprintf("Ignored: %s", res.Error)
```
Comment (yalosev) [inline go_lib/deckhouse-config/status.go:150]: Shouldn't we have an `Error: %s` prefix here? Because we will have always enabled module here

**Review by z9r5 — APPROVED** [review]

File: go_lib/deckhouse-config/status.go (hunk 1 above)
Comment (libmonsoon-dev) [reply go_lib/deckhouse-config/status.go]: Fixed

**Pull request merged**